src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -816,10 +816,41 @@ public void RemoveChangeListener(ListenerToken token)
         public bool Save([NotNull]MutableDocument document, ConcurrencyControl concurrencyControl)
         {
             var doc = CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(document), document);
-            return Save(doc, concurrencyControl, false);
+            return Save(doc, null, concurrencyControl, false);
         }
 
-        #if CBL_LINQ
+        /// <summary>
+        /// Saves a document to the database. When write operations are executed concurrently, 
+        /// and if conflicts occur, conflict handler will be called. Use the handler to directly
+        /// edit the document.Returning true, will save the document. Returning false, will cancel
+        /// the save operation.
+        /// </summary>
+        /// <param name="document">The document to save</param>
+        /// <param name="conflictHandler">The conflict handler block which can be used to resolve it.</param> 
+        /// <returns><c>true</c> if the save succeeded, <c>false</c> if there was a conflict</returns>
+        public bool Save([NotNull]MutableDocument document, Func<MutableDocument, Document, bool> conflictHandler)
+        {
+            var doc = CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(document), document);
+            Debug.Assert(conflictHandler != null);
```

Comment:
Since this comes from outside the library, use `CBDebug.MustNotBeNull` the way that `doc` does.